subworkflows/nf-core/vcf_stats_somalier/main.nf

```diff
@@ -0,0 +1,64 @@
+include { SOMALIER_EXTRACT } from '../../../modules/nf-core/somalier/extract/main'
+include { SOMALIER_RELATE  } from '../../../modules/nf-core/somalier/relate/main'
+include { TABIX_TABIX      } from '../../../modules/nf-core/tabix/tabix/main'
+
+workflow VCF_STATS_SOMALIER {
+    take:
+        ch_vcfs                 // channel: [mandatory] [ meta, vcf, tbi ]
+        ch_fasta                // channel: [mandatory] [ fasta ]
+        ch_fasta_fai            // channel: [mandatory] [ fai ]
+        ch_somalier_sites       // channel: [mandatory] [ somalier_sites_vcf ]
+        ch_peds                 // channel: [optional]  [ meta, ped ]
+        ch_sample_groups        // channel: [optional]  [ txt ]
+    main:
+
+    ch_versions         = Channel.empty()
+
+    ch_input = ch_vcfs
+        .branch { meta, vcf, tbi ->
+            tbi: tbi != []
+                return [ meta, vcf, tbi ]
+            no_tbi: tbi == []
+                return [ meta, vcf ]
+        }
+
+    TABIX_TABIX(
+        ch_input.no_tbi
+    )
+
+    ch_versions = ch_versions.mix(TABIX_TABIX.out.versions)
+
+    ch_somalierextract_input = ch_input.no_tbi
+        .join(TABIX_TABIX.out.tbi)
+        .mix(ch_input.tbi)
+
+    SOMALIER_EXTRACT(
+        ch_somalierextract_input,
+        ch_fasta,
+        ch_fasta_fai,
+        ch_somalier_sites
+    )
+
+    ch_versions = ch_versions.mix(SOMALIER_EXTRACT.out.versions)
+
+    ch_somalierrelate_input = SOMALIER_EXTRACT.out.extract
+        .groupTuple()
+        .join(ch_peds ?: Channel.empty(), by:0 , remainder:true)
+        .map { meta, extract, ped ->
+            [ meta, extract, ped ?: [] ]
+        }.view()
+
+    SOMALIER_RELATE(
+        ch_somalierrelate_input,
+        ch_sample_groups
+    )
+
+    ch_versions = ch_versions.mix(SOMALIER_EXTRACT.out.versions)
+
+    emit:
+    somalier_extract        = SOMALIER_EXTRACT.out.extract
+    somalier_html           = SOMALIER_RELATE.out.html
+    somalier_pairs_tsv      = SOMALIER_RELATE.out.pairs_tsv
+    somalier_samples_tsv    = SOMALIER_RELATE.out.samples_tsv
```

Comment:
> I would maybe rename this slightly, I don't think `stats` is particularly descriptive when the two modules used are called 'extract' and 'relate'. Maybe just `extract_relate` as the middle part?
> 
> Ohteriwse overall LGTM (I can't check the test.yml because GH is borked...'

Good point, it's not only stats indeed, but it will mainly be used for its stats :) But `extract_relate` might be more descriptive indeed.